src/tools/crawling_tools.py

```
@@ -1038,3 +1354,110 @@ async def _update_sources(ctx: Context, source_content_map: Dict[str, str], sour
+    
+    # Content-based classification
+    if content_lower.count('def ') > 5 or content_lower.count('class ') > 3:
+        return "CodeReference"
+    
+    if content_lower.count('step') > 3 or 'getting started' in content_lower:
+        return "Tutorial"
+    
+    return "Documentation"
+
+
+def _extract_sections(content: str) -> List[Dict[str, Any]]:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)